MyCryptoHQ/MyCrypto
# Enhance `View Address` screen with a dropdown that draws from address book and recent addresses

Closes #2044

### Description

Enhances the `View Address` component by replacing the previous recent addresses selection with a dropdown that allows the user to select an entry from their address book _or_ recent addresses.

### Changes

* Old `recent addresses` select was removed.
* New input for address book entries + recent addresses added.
* New optional properties can be passed to `<AddressField />`
* `<AddressFieldDropdown />` now has a connected and unconnected version.

### Steps to Test

1. Visit the `View Address` view from the homepage.
2. Focus the top input.
3. A list of address book entries and recent addresses should appear.
4. Clicking or navigating via keyboard and hitting `Enter` should take you to the `View Wallet (unlocked)` page.
5. The `To` field in `SendTransaction` should behave as before, with the `<AddressField />` having an identicon, etc.

### Screenshots

<img width="501" alt="screen shot 2018-07-12 at 4 20 43 pm" src="https://user-images.githubusercontent.com/10479826/42660203-114ac61e-85f0-11e8-8391-15f45b4c56da.png">
<img width="463" alt="screen shot 2018-07-12 at 4 20 51 pm" src="https://user-images.githubusercontent.com/10479826/42660204-115aea58-85f0-11e8-829e-20808c023394.png">
